stdlib/open-uri.rb

```diff
@@ -0,0 +1,515 @@
+      var xhr = new XMLHttpRequest();
+      xhr.open('GET', uri, false);
+      xhr.send();
+      return xhr;
+    }
+  end
+
+  class HTTPError < StandardError
+    def initialize(message, io)
+      super(message)
+      @io = io
+    end
+    attr_reader :io
+  end
+
+  # Raised on redirection,
+  # only occurs when +redirect+ option for HTTP is +false+.
+  class HTTPRedirect < HTTPError
+    def initialize(message, io, uri)
+      super(message, io)
+      @uri = uri
+    end
+    attr_reader :uri
+  end
+
+  class Buffer # :nodoc: all
+    def initialize
+      @io = StringIO.new
+      @size = 0
+    end
+    attr_reader :size
+
+    def <<(str)
+      @io << str
+      @size += str.length
+    end
+
+    def io
+      Meta.init @io unless Meta === @io
+      @io
+    end
+  end
+
+  # Mixin for holding meta-information.
+  module Meta
+    def Meta.init(obj, src=nil) # :nodoc:
+      obj.extend Meta
+      obj.instance_eval {
+        @base_uri = nil
+        @meta = {} # name to string.  legacy.
+        @metas = {} # name to array of strings.
+      }
+      if src
+        obj.status = src.status
+        obj.base_uri = src.base_uri
+        src.metas.each {|name, values|
+          obj.meta_add_field2(name, values)
+        }
+      end
+    end
+
+    # returns an Array that consists of status code and message.
+    attr_accessor :status
+
+    # returns a URI that is the base of relative URIs in the data.
+    # It may differ from the URI supplied by a user due to redirection.
+    attr_accessor :base_uri
+
+    # returns a Hash that represents header fields.
+    # The Hash keys are downcased for canonicalization.
+    # The Hash values are a field body.
+    # If there are multiple field with same field name,
+    # the field values are concatenated with a comma.
+    attr_reader :meta
+
+    # returns a Hash that represents header fields.
+    # The Hash keys are downcased for canonicalization.
+    # The Hash value are an array of field values.
+    attr_reader :metas
+
+    def meta_setup_encoding # :nodoc:
+      charset = self.charset
+      enc = nil
+      if charset
+        begin
+          enc = Encoding.find(charset)
+        rescue ArgumentError
+        end
+      end
+      enc = Encoding::ASCII_8BIT unless enc
+      if self.respond_to? :force_encoding
+        self.force_encoding(enc)
+      elsif self.respond_to? :string
+        self.string.force_encoding(enc)
+      else # Tempfile
+        self.set_encoding enc
+      end
+    end
+
+    def meta_add_field2(name, values) # :nodoc:
+      name = name.downcase
+      @metas[name] = values
+      @meta[name] = values.join(', ')
+      meta_setup_encoding if name == 'content-type'
+    end
+
+    def meta_add_field(name, value) # :nodoc:
+      meta_add_field2(name, [value])
+    end
+
+    # returns a Time that represents the Last-Modified field.
+    def last_modified
+      if vs = @metas['last-modified']
+        v = vs.join(', ')
+        Time.httpdate(v)
```

Comment:
There's no such method in Opal, so I guess that's a dead code?

Reply:
I've implemented this method, I will add a test